src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier randomColor = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(Figures.values().length);
+        String figureType = String.valueOf(Figures.values()[index]);
+
+        if (figureType.equals(Figures.SQUARE.toString())) {
+            return new Square(randomColor.getRandomColor(),
```

Comment:
we can move creating square to method

Reply:
Do you mean method in this class like getNewSquare which returns object?

Reply:
yes create separate methods to get each figure